src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -494,6 +495,563 @@ class C4 { }";
+                        actualNonNullTypes);
+                    Assert.Equal(new[] {
+                        "System.Runtime.CompilerServices.CompilerGeneratedAttribute",
+                        "Microsoft.CodeAnalysis.EmbeddedAttribute" },
+                        actualEmbedded);
+                }
+                else
+                {
+                    Assert.Equal(new[] {
+                        "System.AttributeUsageAttribute(System.AttributeTargets.Module | System.AttributeTargets.Class | System.AttributeTargets.Struct | System.AttributeTargets.Enum | System.AttributeTargets.Constructor | System.AttributeTargets.Method | System.AttributeTargets.Property | System.AttributeTargets.Field | System.AttributeTargets.Event | System.AttributeTargets.Interface | System.AttributeTargets.Delegate)" },
+                        actualNonNullTypes);
+                    Assert.Empty(actualEmbedded);
+                }
+            };
+
+            CompileAndVerify(comp, sourceSymbolValidator: validator, symbolValidator: validator);
+        }
+
+        [Fact]
+        public void NonNullTypesAttribute_Injected_WithAmbiguousUserDefinedType()
+        {
+            var source = @"
+using System.Runtime.CompilerServices;
+using N1;
+
+[module: NonNullTypesAttribute(true)]
+namespace N1
+{
+    public class NonNullTypesAttribute : System.Attribute
+    {
+        public NonNullTypesAttribute(bool flag) => throw null;
+    }
+}
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (5,10): error CS0104: 'NonNullTypesAttribute' is an ambiguous reference between 'N1.NonNullTypesAttribute' and 'System.Runtime.CompilerServices.NonNullTypesAttribute'
+                // [module: NonNullTypesAttribute(true)]
+                Diagnostic(ErrorCode.ERR_AmbigContext, "NonNullTypesAttribute").WithArguments("NonNullTypesAttribute", "N1.NonNullTypesAttribute", "System.Runtime.CompilerServices.NonNullTypesAttribute").WithLocation(5, 10)
+                );
+        }
+
+        [Fact]
+        public void NonNullTypesAttribute_Injected_WithNonAmbiguousUserDefinedType()
+        {
+            var source = @"
+[module: N1.NonNullTypesAttribute(true)]
+namespace N1
+{
+    public class NonNullTypesAttribute : System.Attribute
+    {
+        public NonNullTypesAttribute(bool flag) => throw null;
+    }
+}
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void NonNullTypesAttribute_WithoutUsage()
+        {
+            var comp = CreateEmptyCompilation("namespace System { public struct Void { } public class Object { } public abstract class ValueType { } }", parseOptions: TestOptions.Regular8);
```

Comment:
>CreateEmptyCompilation("namespace System { public struct Void { } public class Object { } public abstract class ValueType { } }" [](start = 23, length = 128)

Is the empty compilation important? If not, consider using `CreateCompilation("class C { }")` instead.